pkg/cmd/dev/lint.go

```diff
@@ -121,7 +122,15 @@ func (d *dev) lint(cmd *cobra.Command, commandLine []string) error {
 	if err != nil {
 		return err
 	}
-	if !short && filter == "" {
+	if pkg != "" {
+		toLint := strings.TrimPrefix(pkg, "./")
+		args := []string{"build", toLint, "--//build/toolchains:nogo_flag"}
+		if numCPUs != 0 {
+			args = append(args, fmt.Sprintf("--local_cpu_resources=%d", numCPUs))
+		}
+		logCommand("bazel", args...)
+		return d.exec.CommandContextInheritingStdStreams(ctx, "bazel", args...)
+	} else if !short && filter == "" {
```

Comment:
(this question is outside the scope of the PR, but would someone ever want to use a filter and have the linter enabled against that as well?)

Reply:
I think it makes sense that if you provide a filter, then you should be skipping this build step. Presumably in this case you are interested only in the lint checks that match the filter, and this step (which you can't "filter" over) wouldn't match it. I updated the code accordingly.